src/traces/parcoords/calc.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var hasColorscale = require('../../components/colorscale/has_colorscale');
+var calcColorscale = require('../../components/colorscale/calc');
+var Lib = require('../../lib');
+
+
+module.exports = function calc(gd, trace) {
+    var inputDimensions = trace.dimensions,
+        dimensions = [];
+    for(var i = 0; i < inputDimensions.length; i++) {
```

Comment:
Commit https://github.com/plotly/plotly.js/commit/778ebb8aeb2aaeb58d7ee6f4b060a5c23c8b0861 removes this (useless) loop in the calc step. As all calcdata items have a reference to their corresponding `_fullData` traces, it is trivial to find trace setting in `cdparcoords`.